src/components/core/dropdown/dropdown-base/dropdown-base.tsx

```diff
@@ -0,0 +1,654 @@
+import React from 'react';
+import classNames from 'classnames';
+import ChevronDownDark from '@assets/svg/chevron-down-dark.svg';
+import ChevronDownLight from '@assets/svg/chevron-down-light.svg';
+import DisplayText from './display-text';
+import DropdownList from './dropdown-list';
+import { TListItem } from '../types';
+import { useOnClickOutside } from 'hooks';
+import { TDropdownBase, TDirection, TDropdownRef, TDropdownListRef } from '../types';
+import { styled } from 'Styles/stitches.config';
+
+const KEY_CODE = {
+    ENTER: 13,
+    ESCAPE: 27,
+    KEYDOWN: 40,
+    KEYUP: 38,
+    SPACE: 32,
+    TAB: 9,
+};
```

Comment:
```suggestion
const KEY_CODE = {
    ENTER: 13,
    ESCAPE: 27,
    KEYDOWN: 40,
    KEYUP: 38,
    SPACE: 32,
    TAB: 9,
};
```
[Comment1]
KeyboardEvent.KeyCode is deprecated and is no longer recommended to be used: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

Instead you should use KeyBoardEvent.code 
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code